elastic/kibana
# [SecuritySolution] Fix DNS histogram query

## Summary

This fix below query issues within DNS histogram:
### 1. The terms aggregates field should be `dns.question.regustered_domain` instead `dns.question.domain`

**Steps to verify:**
 - Inspect the chart and observe the query should do terms aggregation on `dns.question.regustered_domain`
 
**Known issue:**
https://github.com/elastic/kibana/issues/154533


<img width="1673" alt="Screenshot 2023-04-06 at 12 40 38" src="https://user-images.githubusercontent.com/6295984/230366661-bb9ce5d3-7e80-460e-af64-8fbe26521c1c.png">


### 2. It didn't respect `is PTR record included` flag

**Steps to verify:**
- Switch the toggle in DNS tab to `include PTR record`
- Inspect the chart and observe the query should not exclude PTR record
- Click on Add to new Case / Add to existing Case / Open in Lens and observe the query should not exclude PTR record.



https://user-images.githubusercontent.com/6295984/230366687-8cf0a4dd-beca-46ef-b756-ce898f289c47.mov



### 3. It didn't respect `is PTR record  included` when feature flag `chartEmbeddablesEnabled: is false` 

**Steps to verify:**
- Set chartEmbeddablesEnabled to false
- Switch the toggle in DNS tab to `include PTR record`
- Click on Add to new Case / Add to existing Case / Open in Lens and observe the query should not exclude PTR record.


https://user-images.githubusercontent.com/6295984/230420883-a1541d91-7414-4ce8-9ef5-ffb00e7d174d.mov





**After:**

https://user-images.githubusercontent.com/6295984/230366882-3fe857b0-b3d6-46cf-aaa7-646190ae661f.mov



### Checklist

Delete any items that are not applicable to this PR.
